lib/testing/src/lib/process-services-cloud/pages/task-list-cloud-component.page.ts

```diff
@@ -156,4 +158,29 @@ export class TaskListCloudComponentPage {
         return BrowserActions.getText(locator);
     }
 
+    async clickOptionsButton(content: string) {
+        await BrowserActions.closeMenuAndDialogs();
+        const row: ElementFinder = this.dataTable.getRow('Id', content);
+        await BrowserActions.click(row.element(this.optionButton));
+        await BrowserVisibility.waitUntilElementIsVisible(this.actionMenu);
+    }
+
+    async clickOnCustomActionMenu(action: string): Promise<void> {
+        const actionButton = element(by.css(`button[data-automation-id*="${action}"]`));
+        await BrowserActions.click(actionButton);
+    }
+
+    async isCustomActionEnabled(action: string): Promise<boolean> {
+        const actionButton = element(by.css(`button[data-automation-id*="${action}"]`));
+        return actionButton.isEnabled();
+    }
+
+    async rightClickOnRow(taskId: string): Promise<void> {
+        await this.dataTable.rightClickOnRow('Id', taskId);
+    }
+
+    async clickContextMenuActionNamed(actionName): Promise<void> {
+        await BrowserActions.clickExecuteScript(`button[data-automation-id="context-${actionName}"]`);
+    }
+
```

Comment:
I see the above 5 methods duplicated across the task-list-cloud-component.page.ts and process-list-cloud-component.page.ts. Open to suggestions to make this code better?

Reply:
I think clickOptionsButton can be moved to documentList or datatable PO. clickOnCustomActionMenu(), clickContextMenuActionNamed() and isCustomActionEnabled() can't be part of documentList or datatable, maybe can be part of another PO related to menu. rightClickOnRow() already uses a datatable method, so should stay here.

Reply:
@eromano would you suggest to put the action menu methods in BrowserActions class? or maybe create a separate actionmenu class and add these methods there... pls advise.

Reply:
I don't see this as a big problem as long as we have only 2 places Task and Process